experiments/snippets/skuidCustom/skuidCustom.js

```diff
@@ -0,0 +1,1344 @@
+};
+Date.prototype.getDaysInMonth = function () { 
+	return Date.getDaysInMonth(this.getFullYear(), this.getMonth());
+};
+
+// Date.addMonths(months)
+// Adds a number of months to a date, returns the updated date
+Date.prototype.addMonths = function (months) {
+	const n = this.getDate();
+	this.setDate(1);
+	this.setMonth(this.getMonth() + months);
+	this.setDate(Math.min(n, this.getDaysInMonth()));
+	return this;
+};
+
+// Date.addDays(days)
+// adds a number of days to a date, returns the updated date
+Date.prototype.addDays = function (days) {
+	const date = new Date(this.valueOf());
+	date.setDate(date.getDate() + days);
+	return date;
+};
+
+// Object.byString(ourObject,fieldToGet);
+// Able to get object's sub-objects with a single string variable
+// instead of needing dot notation between multiple separate variables
+// eg: Object.byString(ourObject,'Field__r.SubField__c')
+// instead of needing to do ourObject['Field__r']['SubField__c'] or ourObject.Field__r.SubField__c
+Object.byString = function (ourObject, fieldToGet) {
+	let s = fieldToGet;
+	let o = ourObject;
+	s = s.replace(/\[(\w+)\]/g, '.$1'); // convert indexes to properties
+	s = s.replace(/^\./, ''); // strip a leading dot
+	const a = s.split('.');
+	for (let i = 0, n = a.length; i < n; ++i) {
+		const k = a[i];
+		if (k in o) {
+			o = o[k];
+		} else {
+			return;
+		}
+	}
+	return o;
+};
+
+// skuid.custom.fixCurrency(number)
+// remove anything past 2 digits after the decimal point for a number
+// necessary to do after doing any mathematical operations between 2 floating point numbers
+skuid.custom.fixCurrency = function (num) {
+	// return Math.floor(num*100)/100;
+	return parseFloat(num.toFixed(2));
+};
+
+// skuid.custom.clearArray(array)
+// force-clears the data out of an array which may still otherwise have data in it
+skuid.custom.clearArray = function (array) {
+	while (array.length) {
+		array.pop();
+	}
+};
+
+// skuid.custom.operatorSOQL(operatorSafe)
+// Converts an operator from SKUID XML/HTML safe format to SOQL format
+skuid.custom.operatorSOQL = function (operatorSafe) {
+	if (operatorSafe === 'gt') {
+		return '>';
+	} else if (operatorSafe === 'gte') {
+		return '>=';
+	} else if (operatorSafe === 'lt') {
+		return '<';
+	} else if (operatorSafe === 'lte') {
+		return '<=';
+	}
+
+	return operatorSafe;
+};
+
+// skuid.custom.operatorSafe(operatorSOQL)
+// Converts an operator from SOQL format to SKUID XML/HTML safe format
+skuid.custom.operatorSafe = function (operatorSOQL) {
+	if (operatorSOQL === '>') {
+		return 'gt';
+	} else if (operatorSOQL === '>=') {
+		return 'gte';
+	} else if (operatorSOQL === '<') {
+		return 'lt';
+	} else if (operatorSOQL === '<=') {
+		return 'lte';
+	}
+
+	return operatorSOQL;
+};
+
+// skuid.custom.htmlEscape(string)
+// HTML escape a string, for use when declaring XML for
+// dynamic Template objects that require html escaped contents.
+skuid.custom.htmlEscape = function (string) {

```

Comment:
skuid also has a utility for doing this: `skuid.utils.encodeHTML(string)`

https://docs.skuid.com/latest/v1/en/skuid/api/skuid_utils.html#skuid.utils.encodeHTML